torch/_dynamo/variables/user_defined.py

```diff
@@ -840,16 +840,48 @@ def call_hasattr(self, tx, name: str) -> "VariableTracker":
             return variables.ConstantVariable.create(False)
 
     def odict_getitem(self, tx, key):
+        # This helper function is called for dicts or OrderedDicts, when atleast
+        # one of the keys is not hashable. If all the keys were hashable, the
+        # dict would have already been tracked as ConstDictVariable. Here, we
```

Comment:
Hmm, if this is the invariant, this seems odd - perhaps its time to revisit ODictVariable vs ConstDictVariable vs DefaultDictVariable?